Rubberduck.Main/Root/RubberduckIoCInstaller.cs

```diff
@@ -748,53 +754,29 @@ private void RegisterCustomDeclarationLoadersToParser(IWindsorContainer containe
                 .LifestyleSingleton());
         }
 
-        public static IEnumerable<Assembly> AssembliesToRegister()
+        public IEnumerable<Assembly> AssembliesToRegister()
         {
-            return Assembly.GetExecutingAssembly()
-                .GetReferencedAssemblies()
-                .Where(name => name.FullName.StartsWith("Rubberduck"))
-                .Select(Assembly.Load)
-                .Concat(FindPlugins())
-                // theoretically we shouldn't have anything to register here, but better safe than sorry
-                .Concat(new[] { Assembly.GetExecutingAssembly() });
+            return GetDistinctTransitivelyReferencedAssemblies(Assembly.GetExecutingAssembly(), name => name.FullName.StartsWith("Rubberduck"))
+                //For some reason the inspections assembly is not referenced transitively.
+                .Concat(new [] { Assembly.GetAssembly(typeof(Inspector)) })
+                //Theoretically we shouldn't have anything to register here, but better safe than sorry.
+                .Concat(new[] { Assembly.GetExecutingAssembly() })
+                .Distinct();
         }
 
-        private static IEnumerable<Assembly> FindPlugins()
+        private IEnumerable<Assembly> GetDistinctTransitivelyReferencedAssemblies(Assembly assembly, Predicate<AssemblyName> filterPredicate)
         {
-            var assemblies = new List<Assembly>();
-            var basePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
-
-            // should be loaded already...
-            var inspectionsAssembly = Path.Combine(basePath, "Rubberduck.Inspections.dll");
-            if (File.Exists(inspectionsAssembly))
-            {
-                assemblies.Add(Assembly.LoadFile(inspectionsAssembly));
-            }
-
-            /* Commented out as we don't want to support the plugins right now...
-            var path = Path.Combine(basePath, "Plug-ins");
-            if (!Directory.Exists(path))
-            {
-                Directory.CreateDirectory(path);
-            }
-
-            foreach (var library in Directory.EnumerateFiles(path, "*.dll"))
-            {
-                try
-                {
-                    assemblies.Add(Assembly.LoadFile(library));
-                }
-                catch (Exception)
-                {
-                    // can we log yet?
-                }
-            }
-            */
-
-            return assemblies;
+            var referencedAssemblies = assembly.GetReferencedAssemblies()
+                                                .Where(asmbly => filterPredicate(asmbly))
```

Comment:
this restricts the paths which we take through the dependency graph. That way we only consider Reference paths that contain only `Rubberduck.*`  assemblies.

This implies that transitive Rubberduck References that have an assembly with a different name in between do not get considered. 
That seems fair, considering we want to lock down what we consider for injection, might be useful to make note of that in a comment or XML-Doc